### ISSUE TITLE
helm-3.0.0-beta.5: unable to return any non-zero exit status other than 1 from helm plugins

### ISSUE THREAD

**Issue by mumoshu** [helm/helm#6788]
helm-3.0.0-beta.5 swallows and rounds any non-zero exit status greater than 1 from a helm plugin to `1`.

This, for example, breaks `helm-diff` which relies on helm able to return `2` when `helm-diff` returned `2`.

This is a regression since helm-3.0.0-beta.5. The problem is exactly the same as #6384, which had been fixed in helm-3.0.0-beta.4 via #6385, but was reintroduced to helm-3.0.0-beta.5 via #6341.

---

Output of `helm version`:

```
version.BuildInfo{Version:"v3.0.0-beta.5", GitCommit:"509a8819b7e8dba86b7982c006fce22628aeea28", GitTreeState:"clean", GoVersion:"go1.13.3"}
```

**Comment by karuppiah7890** [comment]
Oops. I guess this could have been avoided if we had a test for the PR which would have been a safety net and would have failed if someone changed behavior

**Comment by mumoshu** [comment]
@karuppiah7890 Definitely.

I'd like to, but am unsure where to add one.

To avoid such regression, what we need to verify is "what `func main()` in cmd/helm/helm.go returns to the os`. Do you have any good idea?

Would you like to change helm's main func to be testable, like allowing to mock/stub any funcs called from it...?

**Comment by karuppiah7890** [comment]
Sorry, no idea. I've to check the code. I haven't read this part of the code. And currently I've lot of other issues / PRs to work on including helm repo's 🙈 , so I would look for help from the community. Probably first timers. But we can't wait for long. v3 stable is going to be out soon in November (probably in the mid). I forgot the date, but it was mentioned in the dev call yesterday. And we'll be having release candidates

**Comment by mumoshu** [comment]
@karuppiah7890 Okay! Anyway, I can submit a fix tonight and keep fixing it until v3 stable is out if you're okay with that for now :)

**Comment by karuppiah7890** [comment]
> keep fixing it until v3 stable is out

? You mean if we can't write tests - keep fixing it when it breaks? We need to know if it breaks first, manually checking it is not feasible. I wouldn't advise it. 

I'll try to help with the tests for the PR. We can also get help from the maintainers. And in previous PRs also, I have refrained from submitting PRs without tests and maintainers also recommend it, an example - https://github.com/helm/helm/pull/6523#issuecomment-536603044

You can also come up with different ideas - anything, preferably something simple to test this - to make sure if someone breaks it, we know.

**Comment by mumoshu** [comment]
@karuppiah7890 All right. I've added the tests.

Please see the last two commints in #6790.

I won't say it's easy to understand at glance(It isn't straight-forward to write/read and that's why I don't usually write tests like that in my other projects).

But once you figure out you'd think it's simple and no better solution exists.

**Comment by adamreese** [comment]
Fixed in #6790
